agent/eventhandler/task_handler_types.go

```diff
@@ -81,6 +81,12 @@ func (event *sendableEvent) taskShouldBeSent() bool {
 		return false
 	}
 
+	// task event should not be sent if status is STOPPED and includes no
+	// container state change events
+	if tevent.Status == api.TaskStopped && len(tevent.Containers) == 0 {
```

Comment:
If I understand the code correctly, this won't work if the task moved directly to stopped eg: have a circular dependency in the task definition. Please correct me if I'm wrong.

Reply:
@richardpen  What are some of the other scenarios where tasks can transition directly to STOPPED?

Reply:
Currently, the one that we have in code is the circular dependency [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_task_engine.go#L533).

Reply:
right. but, what are the other scenarios in which this kind of thing can happen? is that the only one?

Reply:
Checking the code for `emitTaskEvent`, looks like this is the only one scenario today.